src/partition/FileUtils.h

```diff
@@ -34,6 +34,8 @@
 
 namespace lsst::partition {
 
+class ParquetFile;
+
 /// An input file. Safe for use from multiple threads.
 class InputFile {
```

Comment:
You need to declare the destructor of this class as `virtual` since you are adding virtual methods to the class and inheriting from the class:
```
class InputFile {
public:
    virtual ~InputFile();
```